### DISCUSSION TITLE
Matrix chat

### DISCUSSION THREAD

**Discussion by suprohub** [Pumpkin-MC/Pumpkin#479] [Ideas]
### I've searched existing issues and couldn't find a duplicate.

- [X] I confirm this is not a duplicate.

### Is your feature request related to a problem? Please describe.

Matrix chat can be good alternative to current apps like discord

### Describe the solution you'd like

Create matrix chat

**Comment by DataM0del** [comment]
> Hello, we tried to solve the issue.
> 
> This is what we did:
> 
> Added a new field 'matrix' to the ServerLinksConfig struct to support Matrix chat links. Updated the Default implementation to include an empty string for the matrix field.
> 
> You can review changes in this commit: [jacks-sam1010@a3e5c45](https://github.com/jacks-sam1010/Snowiiii-Pumpkin-410/commit/a3e5c45a0d7b681e063c1054698d08de00babac4).
> 
> Caution
> 
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.
> 
> Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

gullible AI moment.
You aren't supposed to add any fields to ServerLinksConfig, since it should only contain fields that the Notchian (vanilla) Minecraft client supports, for other links, you would add them to the custom hashmap.
Anyways, they meant to create a matrix chat for Pumpkin.

**Comment by Commandcracker** [comment]
@jacks-sam1010 

This is not how it works https://github.com/jacks-sam1010/Snowiiii-Pumpkin-410/commit/a3e5c45a0d7b681e063c1054698d08de00babac4.
See https://minecraft.wiki/w/Minecraft_Wiki:Projects/wiki.vg_merge/Protocol#Server_Links_(configuration).
Just use custom if you want to add anything, no changes required.

@suprohub 
Your question is not that precise, did you mean to create an official Matrix group for pumpkin ?
I don't think we will do that because moderating matrix is harder than discord and I don't think we want to have extra moderation effort.

**Comment by suprohub** [comment]
pofig
